dvc/repo/plots/template.py

```diff
@@ -39,6 +43,12 @@ def __init__(self, content=None, name=None):
     def render(self, data, props=None):
         props = props or {}
 
+        if self.METRIC_DATA_ANCHOR not in self.content:
+            raise BadTemplateError(
+                f"Template '{self.filename}' is not using "
+                f"'{self.METRIC_DATA_ANCHOR}' anchor"
+            )
```

Comment:
Wanted to put it somewhere up the chain, but `self.content` might be a dict or a string, because we broke it while refactoring and getting rid of html templates. Looks like we will have to go in and refactor this (the whole plots) again later with the assumption of json and, thus, dicts.